### PR TITLE
[Inference]Update inference config and fix test

### PR DESCRIPTION
**Pull request by CjhHa1**
## 📌 Checklist before creating the PR

- [x] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

fixed #5149 

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.
Update inference config and fix test.
The config.py should be in colossal inference dir , and renamed the inference struct to inferenceData

## 💥 Checklist before requesting a review

- [x] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [x] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

**Comment by CjhHa1** [comment]
image

File: colossalai/inference/config.py [hunk 1]
```diff
@@ -43,8 +45,24 @@ class InferenceConfig:
     max_seq_len: Optional[int] = None
     quant_mode: Optional[str] = None
     revision: Optional[str] = None
-    # TODO: beam search is not support for now
     beam_width: int = 1
+    # TODO: beam search is not support for now
+    ratio: Optional[float] = 1.2
```
Comment (FrankLeeeee) [inline colossalai/inference/config.py:50]: 2 actions need for this arg `ratio`:
1. name it clearly, generally in the form of `xxx_ratio`
2. add its doc to the docstring.

File: colossalai/inference/config.py [hunk 2]
```diff
@@ -43,8 +45,24 @@ class InferenceConfig:
...
+    # the ratio of prefill sequences to decoding sequences, we do prefill step once the actual value exceeds ratio
+
+    def __init_batch_size__(self):
```
Comment (FrankLeeeee) [inline colossalai/inference/config.py:53]: This function is not called?

Comment (FrankLeeeee) [reply colossalai/inference/config.py:53]: `__xxx__` is the format of Python's built-in magic methods, please avoiding naming your function like this. Using `_init_batch_size` will be enough.

File: colossalai/inference/config.py [hunk 3]
```diff
@@ -43,8 +45,24 @@ class InferenceConfig:
...
+        We take a simple method to determine it by GPU memory size, user can still set it manually.
+        """
+        device = torch.device("cuda")
+        total_mem = torch.cuda.get_device_properties(device).total_memory // GibiByte
+        self.max_batch_size = 8
+
+        if 40 < total_mem <= 60:
+            self.max_batch_size = 16
+        elif 60 < total_mem <= 80:
+            self.max_batch_size = 32
```
Comment (FrankLeeeee) [inline colossalai/inference/config.py:65]: Shouldn't this be passed in the by the user?

File: colossalai/inference/inferenceData.py [hunk 4]
```diff
@@ -2,6 +2,10 @@
 from dataclasses import dataclass
 from typing import Dict, List, Set
 
+"""
+The abstraction of request and sequence are defined here.
+"""
```
Comment (FrankLeeeee) [inline colossalai/inference/inferenceData.py:7]: This file's name is not consistent with Python convention, `data.py` will be enough.

File: colossalai/inference/config.py (hunk 3 above)
Comment (CjhHa1) [reply colossalai/inference/config.py:65]: We set this config `max_batch_size` to accurately control memory usage, if a user doesn't know how to set a suitable batch size, they can use this func to set a default value

File: colossalai/inference/inferenceData.py (hunk 4 above)
Comment (CjhHa1) [reply colossalai/inference/inferenceData.py:7]: fixed

File: colossalai/inference/config.py (hunk 1 above)
Comment (CjhHa1) [reply colossalai/inference/config.py:50]: done

File: colossalai/inference/config.py (hunk 2 above)
Comment (CjhHa1) [reply colossalai/inference/config.py:53]: done

File: colossalai/inference/config.py [hunk 5]
```diff
@@ -43,8 +46,28 @@ class InferenceConfig:
     max_seq_len: Optional[int] = None
     quant_mode: Optional[str] = None
     revision: Optional[str] = None
-    # TODO: beam search is not support for now
     beam_width: int = 1
+    # TODO: beam search is not support for now
+    prefill_ratio: Optional[float] = 1.2
+    # the ratio of prefill sequences to decoding sequences, we do prefill step once the actual value exceeds ratio
+
+    def _init_batch_size(self):
```
Comment (FrankLeeeee) [inline colossalai/inference/config.py:54]: I still don't see where this method gets called in your code?

Comment (CjhHa1) [reply colossalai/inference/config.py:54]: fixed

**Review by FrankLeeeee — APPROVED** [review]

**Comment by CjhHa1** [comment]
image

**Pull request merged**